### ISSUE TITLE
Azure RHELv8 image 8-lvm-gen2 (latest) Uyuni Bootstrap is failing 

### ISSUE THREAD

**Issue by RenownTwiz** [uyuni-project/uyuni#8192]
### Problem description

When building an Azure RHEL v8 VM using:
 "imageReference": {
     "publisher": "RedHat",
     "offer": "RHEL",
     "sku": "8-lvm-gen2",
     "version": "latest",
     "exactVersion": "8.9.2023122211"
 },
 
Then running bootstrap.sh from Uyuni 2023.12, the bootstrap completes fine but no repositories are mapped to the client.  And that attached error message shows up in the venv-salt-minion.log and on the Uyuni console.

If you lock down your built to use an older image version (the last pre OELv8 SP9 image):
      "version": "8.8.2023081717",
     "exactVersion": "8.8.2023081717"
It works without issue.

### Steps to reproduce

1.Build Redhat OELv8 "sku": "8-lvm-gen2", "version": "latest" out in Azure
2.Download and run bootstap.sh from Uyuni (Completes successfully)
3.Provided error message is in the venv-salt-minion.log, and shows up on Uyuni console, repositories are not mapped to the client.

### Uyuni version

```bash
Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : Uyuni Server Stable
Name           : Uyuni-Server-release
Version        : 2023.12-230900.210.2.uyuni3
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes
Status         : out-of-date (version 2023.09-230900.207.8.uyuni3 installed)
Source package : Uyuni-Server-release-2023.12-230900.210.2.uyuni3.src
Summary        : Uyuni Server
```

### Useful logs

```bash
Error: Could not parse state file. Please check YAML syntax.
- "Rendering SLS 'base:channels' failed: Jinja error: While reading from '/etc/yum.conf' [line  8]: option 'http_caching' in section 'main' already exists\n\
    /var/cache/venv-salt-minion/files/base/channels/disablelocalrepos.sls(8):\n---\n[...]\n{% if not repos_disabled is defined %}\n{% set repos_disabled\
    \ = {'match_str': 'susemanager:', 'matching': false} %}\n{% endif %}\n{% do repos_disabled.update({'count': 0}) %}\n\n{% set repos = salt['pkg.list_repos']()\
    \ %}    <======================\n{% for alias, data in repos.items() %}\n{% if grains['os_family'] == 'Debian' %}\n{% for entry in data %}\n{% if (repos_disabled.match_str\
    \ in entry['file'])|string == repos_disabled.matching|string and entry.get('enabled', True) %} \ndisable_repo_{{ repos_disabled.count }}:\n[...]\n---\n\
    Traceback (most recent call last):\n  File \"/usr/lib/venv-salt-minion/lib64/python3.10/site-packages/salt/utils/templates.py\", line 477, in render_jinja_tmpl\n\
    \    output = template.render(**decoded_context)\n  File \"/usr/lib/venv-salt-minion/lib64/python3.10/site-packages/jinja2/environment.py\", line 1291,\
    \ in render\n    self.environment.handle_exception()\n  File \"/usr/lib/venv-salt-minion/lib64/python3.10/site-packages/jinja2/environment.py\", line\
    \ 925, in handle_exception\n    raise rewrite_traceback_stack(source=source)\n  File \"
```

### Additional information

This might or might not be important.

Comparing what we have on other VMs that got boostrapped before getting OELv8 sp9, and the latest image out on Azure we do see a few python libraries that are newer out on the Azure build, they might have something to do with the error:

Older VMs bootstrapped on OELv8 prior to SP9  --> From test server built on Azure "latest"
---------
platform-python-3.6.8-56.el8_9.x86_64  --> platform-python-3.6.8-56.el8_9.3.x86_64

python3-libs-3.6.8-56.el8_9.x86_64 --> python3-libs-3.6.8-56.el8_9.3.x86_64

python3-libxml2-2.9.7-16.el8_8.1.x86_64 --> python3-libxml2-2.9.7-18.el8_9.x86_64

python3-perf-4.18.0-513.9.1.el8_9.x86_64 --> python3-perf-4.18.0-513.11.1.el8_9.x86_64

python3-urllib3-1.24.2-5.el8.noarch --> python3-urllib3-1.24.2-5.el8_9.2.noarch

**Comment by avshiliaev** [comment]
Hey @RenownTwiz 
thanks for your report. 
Can you take a look into the `/etc/yum.conf` file? Is the `http_caching` flag is duplicated?

**Comment by RenownTwiz** [comment]
No just created a fresh VM from the template and it looks like this:

[root@jasonvtemp ~]# cat /etc/yum.conf
[main]
gpgcheck=1
installonly_limit=3
clean_requirements_on_remove=True
best=True
skip_if_unavailable=False
http_caching=packages
[root@jasonvtemp ~]#

**Comment by RenownTwiz** [comment]
Sorry posted to quick previously.  Built a test Azure VM and got the wrong template the first time.  I checked this against the "latest" and yes it looks like they have that baked into the template:

[root@jasonvtemp ~]# cat /etc/yum.conf
[main]
gpgcheck=1
installonly_limit=3
clean_requirements_on_remove=True
best=True
skip_if_unavailable=False
http_caching=packages
http_caching=packages
[root@jasonvtemp ~]#

**Comment by RenownTwiz** [comment]
I tested again removing the duplicate line and that is indeed what is causing Uyuni bootstrap to fail.

Looks like that conf file is something in the Azure template, reinstall of the rpm doesn't include it.  From the below Microsoft page it suggests we should be able to open a Microsoft ticket to have them look at this.

https://learn.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-images

I'll see if I can get a Microsoft case open and maybe in the next build of the Azure RHEL v8 template Uyuni bootstrap will start working again with out having to edit the yum.conf file first.

**Comment by RenownTwiz** [comment]
Microsoft has confirmed that they and RedHat have a fix for this and a new Azure image should be released early this week that fixes it.  They also confirm that this is messed up in RHELv8 & RHELv9 images.  As time permits I'll go look to see if they have a new release and test it and confirm back here if it's fixed in RHELv8.

**Comment by RenownTwiz** [comment]
This has been fixed on the Azure side and this issue can be closed.  I've tested using the "latest" Azure RHELv8 image and it now works without any manual changes.  Microsoft has released RHEL/Skus/8-lvm-gen2/Versions/8.9.2024022012 as long as you use that Azure VM image or newer you shouldn't run into this issue (unless they break it again in a future release).

Thanks.

**Comment by RenownTwiz** [comment]
Resolved
